index.js

```diff
@@ -32,17 +32,58 @@
 //. ## Types
 //.
 //. Sanctuary uses Haskell-like type signatures to describe the types of
-//. values, including functions. `'foo'`, for example, has type `String`;
-//. `[1, 2, 3]` has type `Array Number`. The arrow (`->`) is used to express
-//. a function's type. `Math.abs`, for example, has type `Number -> Number`.
-//. That is, it takes an argument of type `Number` and returns a value of
-//. type `Number`.
+//. values, including functions. `'foo'`, for example, is a member of `String`;
+//. `[1, 2, 3]` is a member of `Array Number`. The double colon (`::`) is used
+//. to mean "is a member of", so one could write:
 //.
-//. [`R.map`][R.map] has type `(a -> b) -> Array a -> Array b`. That is,
-//. it takes an argument of type `a -> b` and returns a value of type
-//. `Array a -> Array b`. `a` and `b` are type variables: applying `R.map`
-//. to a value of type `String -> Number` will result in a value of type
-//. `Array String -> Array Number`.
+//.     'foo' :: String
+//.     [1, 2, 3] :: Array Number
+//.
+//. An identifier may appear to the left of the double colon:
+//.
+//.     Math.PI :: Number
+//.
+//. The arrow (`->`) is used to express a function's type:
+//.
+//.     Math.abs :: Number -> Number
+//.
+//. That states that `Math.abs` is a unary function which takes an argument
+//. of type `Number` and returns a value of type `Number`.
+//.
+//. Some functions are parametrically polymorphic: their types are not fixed.
+//. Type variables are used in the representations of such functions:
+//.
+//.     S.I :: a -> a
+//.
+//. `a` is a type variable. Type variables are not capitalized, so they
+//. are differentiable from type identifiers (which are always capitalized).
+//. By convention type variables have single-character names. The signature
+//. above states that `S.I` takes a value of any type and returns a value of
+//. the same type. Some functions require multiple type variables:
+//.
+//.     S.K :: a -> b -> a
+//.
+//. It must be possible to replace all occurrences of `a` with a concrete type.
+//. Ditto for each other type variable. In the case of the function above, the
+//. types to which `a` and `b` resolve may be different, but need not be.
+//.
+//. Since all Sanctuary functions are curried it's common to see a binary
+//. function represented as `* -> * -> *`: a unary function which returns
+//. a unary function. This aligns neatly with Haskell, where all functions
+//. are curried. In JavaScript, though, we may wish to represent the types of
+//. functions with arities less than or greater than one. The general form is
+//. `(<input-types>) -> <output-type>`, where `<input-types>` comprises zero
+//. or more comma–space (<code>, </code>) -separated type representations:
```

Comment:
We need to use `(<code>, </code>)` here because GitHub renders `` (`, `) `` as (`, `) rather than (<code>, </code>).

I first tried the NO-BREAK SPACE character, but our linter complained about [irregular whitespace][1].


[1]: http://eslint.org/docs/rules/no-irregular-whitespace